src/QueryRenderer.ts

```diff
@@ -213,17 +213,14 @@ class QueryRenderChild extends MarkdownRenderChild {
     }: {
         tasks: Task[];
         content: HTMLDivElement;
-    }): Promise<{ taskList: HTMLUListElement; tasksCount: number }> {
-        const tasksCount = tasks.length;
-
+    }): Promise<HTMLUListElement> {
         const layout = new TaskLayout(this.query.layoutOptions);
         const taskList = content.createEl('ul');
         taskList.addClasses(['contains-task-list', 'plugin-tasks-query-result']);
         taskList.addClasses(layout.taskListHiddenClasses);
         const groupingAttribute = this.getGroupingAttribute();
         if (groupingAttribute && groupingAttribute.length > 0) taskList.dataset.taskGroupBy = groupingAttribute;
-        for (let i = 0; i < tasksCount; i++) {
-            const task = tasks[i];
+        for (const [i, task] of tasks.entries()) {
```

Comment:
I'm going to merge this, but for future reference, this change took the PR:
- from 'It's trivial, I'm going to merge it'
- to 'OK, this is gonna be a 10 minute review....'

On the up-side, Arlo's commit notation (`!`) told me to look at it more carefully, so that was really good.

On the other side, I didn't know about `.entries()`, and so had to look it up, and figure out if it had any performance impacts on very large arrays.

Reply:
I do see the benefit of removing the variable for 'number of entries'... so I guess the message is that Arlo's notation really was very useful here - and I learned something!

Reply:
> I'm going to merge this, but for future reference, this change took the PR:
> 
> * from 'It's trivial, I'm going to merge it'
> * to 'OK, this is gonna be a 10 minute review....'
> 

was there another way to do this?

> On the up-side, Arlo's commit notation (`!`) told me to look at it more carefully, so that was really good.

Unfortunately WS doesn't offer `for` loop conversion as an automatic refactor, so I had to do this manually and draw your attention to this.

Reply:
I don't think there was. I tried to get WebStorm to do it, and couldn't.

It's all good, I promise!